addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -2691,47 +2679,59 @@ export class OdooEditor extends EventTarget {
             ev.stopPropagation();
             this.execCommand('strikeThrough');
         } else if (IS_KEYBOARD_EVENT_LEFT_ARROW(ev)) {
-            getDeepRange(this.editable);
             const selection = this.document.getSelection();
+            let range = selection.getRangeAt(0);
+            const isBackwards =
+                !range.collapsed && range.startContainer === selection.focusNode && range.startOffset === selection.focusOffset;
+            range = getDeepRange(this.editable);
             // Find previous character.
-            let { focusNode, focusOffset } = selection;
+            const anchorNode = isBackwards ? range.endContainer : range.startContainer;
+            const anchorOffset = isBackwards ? range.endOffset : range.startOffset;
+            let focusNode = isBackwards ? range.startContainer : range.endContainer;
+            let focusOffset = isBackwards ? range.startOffset : range.endOffset;
```

Comment:
If I do shift+arrow, the result I expect is different based on the direction of the selection.
`a[b]c` -> shift+left-arrow -> `a[]bc`
`a]b[c` -> shift+left-arrow -> `]ab[c`
I would have just used `anchorNode` but `getDeepRange` is needed in order to make the selection deep, and it returns a range, not a selection. And I can't just set it to `select: true` and then retrieve the new selection manually because using `select: true` will trigger a `selectionchange` event (and it's too soon for that, we don't wanna mess it up ^^').

Reply:
```js
// We need to use the range returned by `getDeepRange` but we need
// its direction for shift+arrow (a[b]c -> shift+left -> a[]bc while
// a]b[c -> shift+left -> ]ab[c). Since we can't set
// `getDeepRange`'s `select` option to true then retrieve the
// selection because it would trigger a `selectionchange` event
// before we're done adapting it, we need to compute the direction.
```

Reply:
👍